File: javascript/rna-transcription/rna-transcription.js
class Transcriptor {
  toRna(dna) {
   
    const dnaMap = {
      C : 'G',
      G : 'C',
      A : 'U',
      T : 'A'

    };

    const rna = [];
    const dnaArray = dna.split('');

    for (let i = 0; i < dnaArray.length; i++) {
      if (!dnaMap[dnaArray[i]]) {
        throw new Error('Invalid input DNA.')
      } else {
        rna[i] = dnaMap[dnaArray[i]];
      }
    } 
    return rna.join('');
  } 
  
}

export default Transcriptor;

File: ruby/rna-transcription/rna_transcription.rb


class Complement 
  def self.of_dna(dna)
    rna = ""
 

    for i in 0...dna.length do
      case dna[i]
        when "G" then rna += "C"
        when "C" then rna += "G"
        when "T" then rna += "A"
        when "A" then rna += "U"
      

      end

    end

     return rna

  end
end